File: Data Structures & Algorithms/jump-game-ii/submission-2.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        res = end = farthest = 0

        for i in range(len(nums)-1):
            farthest = max(farthest, i+nums[i])
            if farthest >= len(nums)-1:
                return res+1

            if i == end:
                res += 1
                end = farthest

        return res

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int res = 0, l = 0;
        unordered_map<char,int> uMap;

        for (int r = 0; r < s.size(); ++r){
            auto it = uMap.find(s[r]);
            if (it != uMap.end()){
                l = max(l, it->second + 1);
            }

            uMap[s[r]] = r;
            res = max(res, r-l+1);
        }

        return res;
        
    }
};

